luwak/src/main/java/uk/co/flax/luwak/QueryIndex.java

```diff
@@ -257,7 +261,7 @@ public void doSetNextReader(LeafReaderContext context) throws IOException {
 
         @Override
         public boolean needsScores() {
-            return false;
+            return matcher.needsScores();
```

Comment:
Not sure it's necessary to ever score here (and we switched it back in https://github.com/flaxsearch/luwak/commit/ea704f8fe0a60341f09416550ef7252a3b34528f#diff-67da786c2d60f381a41f43d24e15679d) - if it's false, all tests still pass - so either a test is missing for that case or the scores are not needed for this stage.